static/app/views/automations/components/editConnectedMonitors.tsx

```diff
@@ -1,21 +1,23 @@
-import {Fragment, useState} from 'react';
+import {Fragment} from 'react';
 import styled from '@emotion/styled';
 
 import SearchBar from 'sentry/components/searchBar';
 import {t} from 'sentry/locale';
 import {space} from 'sentry/styles/space';
 import type {Detector} from 'sentry/types/workflowEngine/detectors';
 import ConnectedMonitorsList from 'sentry/views/automations/components/connectedMonitorsList';
+import {useConnectedIds} from 'sentry/views/automations/hooks/utils';
 
-export default function EditConnectedMonitors() {
-  const {monitors, connectedMonitorIds, toggleConnected} = useConnectedMonitors();
+interface Props {
+  key: string;
+}
 
-  const connectedMonitors = monitors.filter(monitor =>
-    connectedMonitorIds.has(monitor.id)
-  );
-  const unconnectedMonitors = monitors.filter(
-    monitor => !connectedMonitorIds.has(monitor.id)
-  );
+export default function EditConnectedMonitors({key}: Props) {
+  const monitors: Detector[] = []; // TODO: Fetch monitors from API
+  const {connectedIds, toggleConnected} = useConnectedIds(key);
```

Comment:
Don't forget to account for `connectedIds` that will be initialized from the existing data. Maybe the hook should accept an object that has `storageKey` and `initialConnectedIds`?

Reply:
gah you're right-- updated here https://github.com/getsentry/sentry/pull/91408/commits/9a964e66eb4747362178e993d4934ee05a3ea251